blueflood-core/src/integration-test/java/com/rackspacecloud/blueflood/io/PreaggregatedMetricsRWIntegrationTest.java

```diff
@@ -0,0 +1,543 @@
+            MetadataCache.getInstance().put(pMetric.getLocator(), MetricMetadata.ROLLUP_TYPE.name().toLowerCase(), RollupType.ENUM.toString());
+
+            // generate gauge
+            BluefloodGaugeRollup gaugeRollup = new BluefloodGaugeRollup()
+                    .withLatest(System.currentTimeMillis(), RAND.nextLong());
+            gaugeRollup.setVariance(RAND.nextDouble());
+            locator = Locator.createLocatorFromPathComponents(tenantId, className + ".my.metric.gauge." + RAND.nextInt(10));
+            pMetric = new PreaggregatedMetric(System.currentTimeMillis(), locator, new TimeValue(1, TimeUnit.DAYS), gaugeRollup);
+            expectedLocatorMetricMap.put(locator, pMetric);
+            MetadataCache.getInstance().put(locator, MetricMetadata.TYPE.name().toLowerCase(), null);
+            MetadataCache.getInstance().put(locator, MetricMetadata.ROLLUP_TYPE.name().toLowerCase(), RollupType.GAUGE.toString());
+
+            // generate set
+            Set aSet = Sets.newHashSet(RAND.nextInt(), RAND.nextInt(), RAND.nextInt());
+            BluefloodSetRollup setRollup = new BluefloodSetRollup().withObject(aSet);
+            locator = Locator.createLocatorFromPathComponents(tenantId, className + ".my.metric.set." + RAND.nextInt(10));
+            pMetric = new PreaggregatedMetric(System.currentTimeMillis(), locator, new TimeValue(1, TimeUnit.DAYS), setRollup);
+            expectedLocatorMetricMap.put(locator, pMetric);
+            MetadataCache.getInstance().put(locator, MetricMetadata.TYPE.name().toLowerCase(), null);
+            MetadataCache.getInstance().put(locator, MetricMetadata.ROLLUP_TYPE.name().toLowerCase(), RollupType.SET.toString());
+
+            // generate timer
+            BluefloodTimerRollup timerRollup = new BluefloodTimerRollup()
+                    .withSampleCount(RAND.nextInt(10))
+                    .withSum(RAND.nextDouble())
+                    .withCountPS(RAND.nextDouble())
+                    .withAverage(RAND.nextLong())
+                    .withVariance(RAND.nextDouble())
+                    .withMinValue(RAND.nextInt())
+                    .withMaxValue(RAND.nextInt())
+                    .withCount(RAND.nextInt(200));
+            locator = Locator.createLocatorFromPathComponents(tenantId, className + ".my.metric.timer." + RAND.nextInt(10));
+            timerMetric = new PreaggregatedMetric(System.currentTimeMillis(), locator, new TimeValue(1, TimeUnit.DAYS), timerRollup);
+            expectedLocatorMetricMap.put(locator, timerMetric);
+            MetadataCache.getInstance().put(locator, MetricMetadata.TYPE.name().toLowerCase(), null);
+            MetadataCache.getInstance().put(locator, MetricMetadata.ROLLUP_TYPE.name().toLowerCase(), RollupType.TIMER.toString());
+        }
+    }
+
+    /**
+     * This method is to supply the granularity parameter to some test methods below
+     *
+     * @return
+     */
+    protected Object getGranularitiesToTest() {
```

Comment:
This method is being called by the junitparams, when you annotate a test method like this:
```
@Parameters(method = "getGranularitiesToTest")
```
The junitparams will then iterate through the array and pass in the granularity as a parameter to the test method. I think it's good to control the input within the test method.